shoop/admin/modules/orders/views/detail.py

```diff
@@ -99,10 +99,11 @@ def post(self, request, *args, **kwargs):
         old_status = order.status
         if new_status.role == OrderStatusRole.COMPLETE and not order.can_set_complete():
             raise Problem(_("Unable to set order as completed at this point"))
-        if new_status.role == OrderStatusRole.CANCELED and not order.can_set_canceled():
-            raise Problem(_("Paid, shipped, or canceled orders cannot be canceled"))
-        else:
-            order.set_canceled()
+        if new_status.role == OrderStatusRole.CANCELED:
+            if not order.can_set_canceled():
+                raise Problem(_("Paid, shipped, or canceled orders cannot be canceled"))
+            else:
+                order.set_canceled()
         order.status = new_status
```

Comment:
However, this line reverses that status change and sets it to the new status. Again, for example if new_status was "complete", the order would set to cancelled above then set to "complete" here, ultimately getting the same result for that case (despite being switched to "canceled" temprarily), and since we're currently only handling "complete" or "canceled" as statuses, there really aren't going to be any errors produced.